### DISCUSSION TITLE
server模型推理出错（core dumped）

### DISCUSSION THREAD

**Discussion by wanggp01** [PaddlePaddle/PaddleOCR#12958] [Q&A]
#### 问题描述 / Problem Description
使用ch_PP-OCRv4_det_infer模型进行推理时是正常的，但是使用server模型推理时，就报core dumped：

#### 运行环境 / Runtime Environment
- OS: ubuntu18.04
- Paddle: 2.1.3
- PaddleOCR: release/2.7.1

#### 复现代码 / Reproduction Code
使用的指令：python tools/infer/predict_system.py --image_dir="./tests/1.jpg" --det_model_dir="./models/server_model/ch_PP-OCRv4_det_server_infer/" --rec_model_dir="./models/server_model/ch_PP-OCRv4_rec_server_infer/" --cls_model_dir="models/server_model/ch_ppocr_mobile_v2.0_cls_infer" --use_angle_cls=true
#### 完整报错 / Complete Error Message
--------------------------------------
C++ Traceback (most recent call last):
--------------------------------------
0   paddle_infer::Predictor::Predictor(paddle::AnalysisConfig const&)
1   std::unique_ptr<paddle::PaddlePredictor, std::default_delete<paddle::PaddlePredictor> > paddle::CreatePaddlePredictor<paddle::AnalysisConfig, (paddle::PaddleEngineKind)2>(paddle::AnalysisConfig const&)
2   paddle::AnalysisPredictor::Init(std::shared_ptr<paddle::framework::Scope> const&, std::shared_ptr<paddle::framework::ProgramDesc> const&)
3   paddle::AnalysisPredictor::PrepareProgram(std::shared_ptr<paddle::framework::ProgramDesc> const&)
4   paddle::AnalysisPredictor::OptimizeInferenceProgram()
5   paddle::inference::analysis::Analyzer::RunAnalysis(paddle::inference::analysis::Argument*)
6   paddle::inference::analysis::IrAnalysisPass::RunImpl(paddle::inference::analysis::Argument*)
7   paddle::inference::analysis::IRPassManager::Apply(std::unique_ptr<paddle::framework::ir::Graph, std::default_delete<paddle::framework::ir::Graph> >)
8   paddle::framework::ir::Pass::Apply(paddle::framework::ir::Graph*) const
9   paddle::framework::ir::ConvEltwiseAddBNFusePass::ApplyImpl(paddle::framework::ir::Graph*) const
10  paddle::framework::ir::GraphPatternDetector::operator()(paddle::framework::ir::Graph*, std::function<void (std::map<paddle::framework::ir::PDNode*, paddle::framework::ir::Node*, std::less<paddle::framework::ir::PDNode*>, std::allocator<std::pair<paddle::framework::ir::PDNode* const, paddle::framework::ir::Node*> > > const&, paddle::framework::ir::Graph*)>)
11  paddle::framework::LoDTensor* paddle::framework::Variable::GetMutable<paddle::framework::LoDTensor>()
12  paddle::framework::SignalHandle(char const*, int)
13  paddle::platform::GetCurrentTraceBackString[abi:cxx11]()

----------------------
Error Message Summary:
----------------------
FatalError: `Segmentation fault` is detected by the operating system.
  [TimeInfo: *** Aborted at 1716639975 (unix time) try "date -d @1716639975" if you are using GNU date ***]
  [SignalInfo: *** SIGSEGV (@0x0) received by PID 16460 (TID 0x7f98e8477080) from PID 0 ***]

Segmentation fault (core dumped)

**Comment by UserWangZz** [comment]
考虑是否paddlepaddle-gpu与cuda版本是否匹配

**Comment by wanggp01** [comment]
> 考虑是否paddlepaddle-gpu与cuda版本是否匹配

目前cuda版本：10.1，cudnn版本：7.6.5， paddlepaddle-gpu版本：2.1.3

**Comment by GreatV** [comment]
安装 develop 版本 的 paddlepaddle 试试看 https://www.paddlepaddle.org.cn/

**Comment by pisce188** [comment]
2.8.0的paddleocr cpu运行推理也有类似报错
`
--------------------------------------
C++ Traceback (most recent call last):
--------------------------------------
0   paddle::AnalysisPredictor::ZeroCopyRun()
1   paddle::framework::NaiveExecutor::Run()
2   paddle::framework::OperatorBase::Run(paddle::framework::Scope const&, phi::Place const&)
3   paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, phi::Place const&) const
4   paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, phi::Place const&, paddle::framework::RuntimeContext*) const
5   paddle::framework::OperatorWithKernel::PrepareData(paddle::framework::Scope const&, phi::KernelKey const&, std::vector<std::string, std::allocator<std::string > >*, paddle::framework::RuntimeContext*, phi::Place const&) const

----------------------
Error Message Summary:
----------------------
FatalError: `Segmentation fault` is detected by the operating system.
  [TimeInfo: *** Aborted at 1742955741 (unix time) try "date -d @1742955741" if you are using GNU date ***]
  [SignalInfo: *** SIGSEGV (@0x0) received by PID 1 (TID 0x7fbc91e94700) from PID 0 ***]

`
